
This enables PCIe2 and PCIe3 on Rock5-a and Rock5-b boards

diff --speed-large-files --no-dereference --minimal -Naur linux-6.2.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts linux-6.2.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
--- linux-6.2.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2023-03-09 09:17:38.483181866 +0100
+++ linux-6.2.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2023-03-09 09:15:58.633185539 +0100
@@ -51,6 +51,39 @@
 		vin-supply = <&vcc5v0_sys>;
 	};
 
+	vcc3v3_pcie2x1l0: vcc3v3-pcie2x1l0 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc3v3_pcie2x1l0";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		enable-active-high;
+		regulator-boot-on;
+		regulator-always-on;
+		gpios = <&gpio1 RK_PD2 GPIO_ACTIVE_HIGH>;
+		startup-delay-us = <50000>;
+		vin-supply = <&vcc5v0_sys>;
+	};
+
+	vcc3v3_pcie2x1l2: vcc3v3-pcie2x1l2 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc3v3_pcie2x1l2";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		startup-delay-us = <5000>;
+		vin-supply = <&vcc_3v3_s3>;
+	};
+
+	vcc3v3_pcie30: vcc3v3-pcie30 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc3v3_pcie30";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		enable-active-high;
+		gpios = <&gpio1 RK_PA4 GPIO_ACTIVE_HIGH>;
+		startup-delay-us = <5000>;
+		vin-supply = <&vcc5v0_sys>;
+	};
+
 	leds {
 		compatible = "gpio-leds";
 
@@ -237,6 +283,36 @@
 	status = "okay";
 };
 
+&pcie2x1l2 {
+	reset-gpios = <&gpio3 RK_PB0 GPIO_ACTIVE_HIGH>;
+	vpcie3v3-supply = <&vcc3v3_pcie2x1l2>;
+	status = "okay";
+};
+
+&pcie2x1l0 {
+	reset-gpios = <&gpio4 RK_PA5 GPIO_ACTIVE_HIGH>;
+	vpcie3v3-supply = <&vcc3v3_pcie2x1l0>;
+	status = "okay";
+};
+
+&pcie3x4 {
+	reset-gpios = <&gpio4 RK_PB6 GPIO_ACTIVE_HIGH>;
+	vpcie3v3-supply = <&vcc3v3_pcie30>;
+	status = "okay";
+};
+
+&combphy0 {
+	status = "okay";
+};
+
+&combphy1 {
+	status = "okay";
+};
+
+&pcie30phy {
+	status = "okay";
+};
+
 &pinctrl {
 	leds {
 		led_user_en: led_user_en {
diff --speed-large-files --no-dereference --minimal -Naur linux-6.2.2/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts linux-6.2.2/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
--- linux-6.2.2/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2023-03-09 09:17:38.176515213 +0100
+++ linux-6.2.2/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2023-03-09 09:01:12.052315449 +0100
@@ -51,6 +51,15 @@
 		vin-supply = <&vcc5v0_sys>;
 	};
 
+	vcc3v3_pcie2x1l2: vcc3v3-pcie2x1l2 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc3v3_pcie2x1l2";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		startup-delay-us = <5000>;
+		vin-supply = <&vcc5v0_sys>;
+	};
+
 	leds {
 		compatible = "gpio-leds";
 
@@ -197,3 +206,13 @@
 	pinctrl-0 = <&sdmmc_bus4 &sdmmc_clk &sdmmc_cmd &sdmmc_det>;
 	status = "okay";
 };
+
+&pcie2x1l2 {
+	status = "okay";
+	reset-gpios = <&gpio3 RK_PD1 GPIO_ACTIVE_HIGH>;
+	vpcie3v3-supply = <&vcc3v3_pcie2x1l2>;
+};
+
+&combphy0 {
+	status = "okay";
+};
